pymaster/field.py

```diff
@@ -658,3 +669,104 @@ def get_ell_sampling(self):
         """
         ells = lib.get_ell_sampling_flat(self.fl, int(self.fl.fs.n_ell))
         return ells
+
+
+class NmtFieldCatalog(NmtField):
+    """
+    """
+    def __init__(self, positions, weights, field, lmax, lmax_mask=-1,
+                 spin=None, beam=None, field_is_weighted=False, lonlat=False):
+        # 0. Preliminary initializations
+        if ut.HAVE_DUCC:
+            self.sht_calculator = 'ducc'
+        else:
+            raise ValueError("DUCC is needed, but currently not installed.")
+
+        # These first attributes are compulsory for all fields
+        self.lite = True
+        self.mask = None
+        self.beam = None
+        self.n_iter = None
+        self.n_iter_mask = None
+        self.pure_e = False
+        self.pure_b = False
+        self._Nw = 0.
+        self._Nf = 0.
+        self.ainfo = ut.NmtAlmInfo(lmax)
+
+        # The remaining attributes are only required for non-lite maps
+        self.maps = None
+        self.temp = None
+        self.alm_temp = None
+        self.minfo = None
+        self.n_temp = 0
+
+        # Sanity checks on positions and weights
+        positions = np.array(positions, dtype=np.float64)
+        weights = np.array(weights, dtype=np.float64)
+        if np.shape(positions) != (2, len(weights)):
+            raise ValueError("Positions must be 2D array of shape"
+                             " (2, len(weights)).")
+        if lonlat:
+            positions[0] = np.radians(90. - positions[1])
+            positions[1] = np.radians(positions[0])
+        if not (np.logical_and(positions[0] >= 0.,
+                               positions[0] <= np.pi)).all():
+            if lonlat:
+                raise ValueError("Second dimension of positions must be "
+                                 "latitude in degrees, between -90 and 90.")
+            else:
+                raise ValueError("First dimension of positions must be "
+                                 "colatitude in radians, between 0 and pi.")
+
+        # Compute mask shot noise
+        self._Nw = np.sum(weights**2.)/(4.*np.pi)
+
+        # 1. Compute mask alms and beam
+        # Sanity checks
+        if lmax_mask <= 0:
+            lmax_mask = 4*lmax
```

Comment:
Why is this a good choice for the default `lmax_mask`?

Reply:
This choice is ad hoc, but the resulting mode coupling yields unbiased mean shear power spectra for 500 KiDS-like simulations at `nside_hires=4096`, while being reasonably quick to compute. A lower `lmax_max/lmax=2` or `3` might do the job as well, but this was not tested.

Reply:
Hmmm, I would suggest we keep the default at `lmax_mask=lmax`, since that may be the most obvious default option, and users tend not to read the documentation. Also, didn't we realise that in fact the maximum ell that the mask PCL can go to is `2*lmax`?

Reply:
(and we can recommend enlarging that in the docstring)

Reply:
Fine with me!